### PR TITLE
Feature/Filter-Results-ak-crystal

### PR DESCRIPTION
**Pull request by Akshith-Patil**
## Proposed changes

- Added two dropdown filter menu for Image & Video results on reportPage
- Allows you to filter results by defect type 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Before Image/Video Filter
Before Filter
Before Video Filter

During Image/Video Filter
During Image Filter
During video Filter

After Image/Video
After 1 Filter
After video filter

### CODE REVIEW

File: src/components/reportPage/ReportPage.jsx [hunk 1]
```diff
@@ -24,6 +27,33 @@ const ReportPage = (props) => {
...
+    { value: "8", label: "Missing elevation" },
+    { value: "2", label: "Missing scrim" },
+    { value: "9", label: "Modal sheet missing scrim" },
+    { value: "6", label: "Moving cards behind other cards" },
+    { value: "1", label: "No defect" },
+    { value: "Null value", label: "Null Value" },
+    { value: "3", label: "Snackbar blocking bottom navigation" },
+    { value: "4", label: "Stacking multiple banners" },
+    { value: "7", label: "Stacking multiple snackbars" },
+  ];
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:49]: Could you order the video defects by the value please? And seperate the image and video defects, something like this:
```
const videoDefects = [
{ value: "1", label: "No defect" },
{ value: "2", label: "Missing scrim" },
...
];

const imageDefects = [
{ value: "Component occlusion", label: "Component Occlusion" },
...
];

const allDefects = [...videoDefects, ...imageDefects];
```

File: src/components/reportPage/ReportPage.jsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+import Select from "react-select";
+import makeAnimated from "react-select/animated";
 
 import "./results-page.css";
 import Repository from "../../data/Repository";
 import VideoResult from "./VideoResult";
 import ImageResult from "./ImageResult";
 import { inferno256 } from "./gradients256";
 import ColourSchemeSelector from "./ColourSchemeSelector";
+import { alignPropType } from "react-bootstrap/esm/DropdownMenu";
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:14]: Unused import

File: src/components/reportPage/ReportPage.jsx [hunk 3]
```diff
@@ -24,6 +27,33 @@ const ReportPage = (props) => {
...
+    { value: "3", label: "Snackbar blocking bottom navigation" },
+    { value: "4", label: "Stacking multiple banners" },
+    { value: "7", label: "Stacking multiple snackbars" },
+  ];
+
+  const filterRef = useRef();
+  filterRef.current = filterType;
+
+  const vidFilterRef = useRef();
+  vidFilterRef.current = videoFilterType;
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:55]: I can't see these being used anywhere

File: src/components/reportPage/ReportPage.jsx [hunk 4]
```diff
@@ -24,6 +27,33 @@ const ReportPage = (props) => {
   const [colourScheme, setColourScheme] = useState(inferno256);
   const [searchTerm, setSearchString] = useState("");
 
+  const [filterType, setFilterType] = useState("All");
+  const [videoFilterType, setVideoFilterType] = useState("0");
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:31]: Can't see these being used anywhere

File: src/components/reportPage/ReportPage.jsx [hunk 5]
```diff
@@ -24,6 +27,33 @@ const ReportPage = (props) => {
...
+
+  const [displayValue, getValue] = useState([]);
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:33]: Could we think of a better name for this? It's not very clear what this list is. Also should be `set` not `get`.

File: src/components/reportPage/ReportPage.jsx [hunk 6]
```diff
@@ -51,6 +81,33 @@ const ReportPage = (props) => {
...
+
+    if (
+      displayValue.length === 0 ||
+      (displayValue.includes("1") && imageResult["bug_type"].length === 0) ||
+      found
+    ) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:99]: Could be shortened to `return displayValue.length === 0 ||
      (displayValue.includes("1") && imageResult["bug_type"].length === 0) ||
      found`

File: src/components/reportPage/ReportPage.jsx [hunk 7]
```diff
@@ -51,6 +81,33 @@ const ReportPage = (props) => {
...
+  };
+
+  const checkVideoFilterType = (videoResult) => {
+    const found = displayValue.includes(videoResult["classification"]);
+    if (displayValue.length === 0 || found) {
+      return true;
+    } else {
+      return false;
+    }
+  };
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:109]: Could be shortened to `return displayValue.length === 0 || found`

File: src/components/reportPage/ReportPage.jsx [hunk 8]
```diff
@@ -51,6 +81,33 @@ const ReportPage = (props) => {
...
+  const handleMenuChange = (e) => {
+    getValue(Array.isArray(e) ? e.map((x) => x.value) : []);
+  };
+
+  const checkImageFilterType = (imageResult) => {
+    const found = displayValue.some((value) => imageResult["bug_type"].includes(value));
+
+    if (
+      displayValue.length === 0 ||
+      (displayValue.includes("1") && imageResult["bug_type"].length === 0) ||
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:93]: Maybe define "1" as a constant to give it meaning? `const IMAGE_NO_DEFECT_VALUE = "1";`
`displayValue.includes(IMAGE_NO_DEFECT_VALUE)`

**Review by idomo1 — CHANGES_REQUESTED** [review]
Nice work guys 👍 
Mostly just some minor things that could be improved.

It's a bit misleading where the defect filter is placed. It's under the image results section, but it's applying to both
image

File: src/components/reportPage/ReportPage.jsx (hunk 1 above)
Comment (idomo1) [reply src/components/reportPage/ReportPage.jsx:49]: Just noticed, "1" is being used as no defects for images. There is a "1" for video results though which is missing. Ig you have a look at Repository.js there is a mapping for the video results
image

**Comment by Akshith-Patil** [comment]
Hey @idomo1 Damien, thanks for the feedback, made all changes mentioned above. Had a question regarding the 'classification', if no defect is found, is the 'classification' tag just empty?

**Comment by idomo1** [comment]
@Akshith-Patil From what I understand, for video results "Can't place in space" = No defect. Classification will always be one of the 10 options

File: src/components/reportPage/ReportPage.jsx [hunk 9]
```diff
@@ -24,6 +26,31 @@ const ReportPage = (props) => {
   const [colourScheme, setColourScheme] = useState(inferno256);
   const [searchTerm, setSearchString] = useState("");
 
+  const [selectedImageDefects, setImageDefects] = useState([]);
+  const [selectedVideoDefects, setVideoFefects] = useState([]);
```
Comment (idomo1) [inline src/components/reportPage/ReportPage.jsx:30]: Typo, "setVideoDefects"

**Review by idomo1 — APPROVED** [review]
Just a typo to fix if you can be bothered, no biggy

**Pull request merged**